### DISCUSSION TITLE
Another Mini Hotplate but with 56x56 mm Area

### DISCUSSION THREAD

**Discussion by Montvydas** [Ralim/IronOS#1698] [Q&A]
Has anyone tried taking apart one of those cheap PD 65W mini hot plates from [Aliexpress](https://www.aliexpress.com/item/1005005549693756.html?spm=a2g0o.order_list.order_list_main.5.5c8a1802ZW1TEa) and see if it uses stm32. There is a youtube video [here](https://www.youtube.com/watch?v=cGCsakarCkQ) of a similar one if Aliexpress link doesn't work. It would be cool to see IronOS running on it, since you can buy it for as low as £10 and I really like the bigger surface area of 56x56 mm as opposed to the 30x30 on MHP30 😄

**Comment by Ralim** [comment]
If anyone tears one down and looks whats inside I would be curious. Definitely possible to add if hardware supports it :)

**Comment by StevenSeifried** [comment]
Sounds interesting. I could not resist and bought one for 19 Euros, should have arrived by the end of this month (12 day delivery to Germany). 

When it arrives, I will tear it down. @Ralim do you want the pictures of the PCB here in this discussion or in a separate issue?

**Reply by Ralim** [reply]
Here is good, I've ordered one too but slow shipping (2-8 weeks) so you should beat mine handily.

**Reply by janowakps** [reply]
@Ralim any updates?

**Reply by Ralim** [reply]
It was discussed somewhere else, but basically the chipset in these cant run IronOS so it wont be ported for them.

**Comment by Montvydas** [comment]
I ordered one as well, it was hella cheap and it even includes the PD charger (I hope it won't break on first try!) for £16 with taxes included 😄 I will likely tear the hot plate apart just to ensure it's electrically safe to use, it seems to have this "easy to disassemble" vibe to it.

**Comment by StevenSeifried** [comment]
Mine arrived today. PCB is equipped on one side
- [STC 8H3K64S2](https://www.stcmicro.com/datasheet/STC8H-en.pdf)
- [NCE6050KA](https://datasheet.lcsc.com/lcsc/1809042027_Wuxi-NCE-Power-Semiconductor-NCE6050KA_C96013.pdf)
- [HDM CH224K](https://www.laskakit.cz/user/related_files/ch224ds1.pdf)

**Reply by paulfertser** [reply]
The photo would still nice to have here so that the others would be able to see how it's built, whether it looks safe etc. Thank you for sharing!

**Reply by StevenSeifried** [reply]
![20230616_133523.jpg](https://github.com/Ralim/IronOS/assets/39765956/44f7dd52-0abb-4dab-9d1b-9788aed73c85)

![20230616_130208.jpg](https://github.com/Ralim/IronOS/assets/39765956/3490ab53-f32c-4ee3-8cbf-aa9ea28a717b)

![20230616_130133.jpg](https://github.com/Ralim/IronOS/assets/39765956/7853598b-869f-4432-b4ee-4b1e448da0b4)

**Reply by gamelaster** [reply]
haha I guess this will not be able to run IronOS at all :rofl:

**Reply by StevenSeifried** [reply]
Did we really don't have any chance, that IronOS will run on it?

**Reply by gamelaster** [reply]
@StevenSeifried Let's wait for Ralim, but from my knowledge, it would needed to be extremely trimmed down version of IronOS

**Reply by StevenSeifried** [reply]
The device doesn't have and need (imo) many features. Maybe they can be adopted to IronOS. Current features are:

- Left button -> temperature down
- Right button -> temperature up
- Both buttons -> Start/Stop heating 

As features from IronOS I would be happy about (if possible)
- Long press on the right button opens the settings menu
- Screen brightness 
- advanced idle screen
- advanced heating screen

Flashing IronOS will not be user friendly ootb, since the device doesn't expose UART to the USB-C port. It only has the four pads for UART on the PCB. So, users have to disassemble it and solder it.

A few sellers on AliExpress named it "MHP30", I had a bit of hope that it's just a "low quality" MHP30 clone with a bigger heat plate and without the anti-tilt, aluminum case and RGB led.

We will see what is possible, I'll wait for further instructions.

**Reply by paulfertser** [reply]
8051 architecture, 64 KiB flash, 3 KiB SRAM. Does this really leave a chance?

**Reply by paulfertser** [reply]
Apparently it can be flashed directly over USB though.

**Reply by paulfertser** [reply]
No, this one isn't USB-capable, so UART only. Shouldn't be a complicated process if one writes a replacement firmware. But reading out the original firmware isn't possible (at least no method known to the general public is available).

**Reply by ia** [reply]
Hello. Sorry for the late reply and bothering with off-topic but I'm just curious... could you tell me, please: did you make these photos with photo camera & macro-lens or was it one of those digital microscopes? And in case if it was the latter, could you type the full name of the brand/model? (I was just thinking about getting one of those for myself) Thanks.

**Reply by StevenSeifried** [reply]
I used my smartphone (Samsung Galaxy A52s 5G) to take the photos

**Comment by paulfertser** [comment]
So be it. Let's hope it's not too bad even without IronOS. Please post your review after you get some experience with it here.

**Comment by StevenSeifried** [comment]
I'm unable to get more than 215-220 °C out of it (according to the current temperature shown on display). The temperature was set to 250 °C. Tested with three different 65W USB-PD chargers and two different USB-C-to-USB-C cables (100W and 240W)

**Comment by Montvydas** [comment]
That is strange, since I managed to reach 247C after setting it for 250C (when it's ~5C to the set value it stops heating up and just maintains the temperature while very slowly approaching the set value) using Macbook 100W brick + original cable in just about 2min 30 sec. In addition, the hot plate managed to reach 200C in 1 min 10sec and 220C in just about 1 min 40 sec, which is pretty decent. I guess you wouldn't use it for standard non-leaded solder paste, but it would work great for low-temp non-leaded or leaded pastes.

Edit: Could it be that you need more than 65W to reach above 220C maybe?

**Comment by paulfertser** [comment]
Using a 20 V 3.25 A 65 W PD source I got to 220 degrees in 2.5 minutes, 230 in 3 minutes, 240 in 4.5 minutes. Around 245 was its maximum.

**Comment by Rixxxi** [comment]
I orderd one of these today. Looking forward to my first ever reflow soldering with this little thing.


> The device doesn't have and need (imo) many features. Maybe they can be adopted to IronOS. Current features are:
>      * Left button -> temperature down
>      * Right button -> temperature up
>      * Both buttons -> Start/Stop heating

Have you noticed there is a menu including PID settings?
The menu options are explained in this video: https://youtu.be/3ig7yCUP1xg


> I'm unable to get more than 215-220 °C out of it (according to the current temperature shown on display). The temperature was set to 250 °C. Tested with three different 65W USB-PD chargers and two different USB-C-to-USB-C cables (100W and 240W)

Maybe changing the PID settings could make it reach the max temperature?

**Comment by Blaster1920** [comment]
Just got one myself.
Looks like the stable pid values for me are:
KP 112
KI 004
KD 160
This thing desperately needs and autotuning sequence.

**Reply by Rixxxi** [reply]
Awesome! Thanks for posting your values!

I was using the hot plate yesterday and this morning I wondered how to tune the PID values. I have no idea.
What process you used to figure out which values work stable?

**Comment by Blaster1920** [comment]
Didn't use any particular process, but:
Ki wasn't doing much do I disabled it.
I figured out it wasn't looking enough far ahead in time so I increased Kd.
Tried to get Kp and Kd well balanced to avoid oscillations.
Now the marginal offset error was too high, -2degrees, activated Ki and
played a bit with it.

This is all after trying Nicholson's, and failing in doing so.

With a proper autotune you might get even closer to perfection.

I performed all the testing with a 1.6mm fr4 board unpopulated but with
solder on top, with setpoint at 170° since I'll be using this plate mostly
for flexible pcbs.

Suggestion: for very thin/flexible/super small PCBs, you might want to tune
down Kd slightly, say by 5 or 10.

**Comment by krunchiekrunch** [comment]
Just found out this hot plate is entirely open sourced, including the code and hardware
https://oshwlab.com/oshwhub.com/pd-protocol-high-definition-mini

**Comment by gxcreator** [comment]
If someone wants to try, I've ported original FW to SDCC and refined it here: https://github.com/gxcreator/g3061-hotplate-fw

**Comment by elektroencefalografista** [comment]
The CH32V003 in TSSOP-20 package is ALMOST pin-compatible with the controller they used, at least in my version (CMS8S5897). I had to cut and move SDA trace to PC1, and also cut and move the NTC trace somewhere else where the ADC maps. The power pins, buttons, MOSFET and VIN sensing all land on adequate pads.
